Sources/TuistGraph/Models/DeploymentTargets.swift

```diff
@@ -0,0 +1,73 @@
+import Foundation
+
+// MARK: - DeploymentTargets
+
+public struct DeploymentTargets: Hashable, Codable {
+    
+    public let iOS: String?
+    public let macOS: String?
+    public let watchOS: String?
+    public let tvOS: String?
```

Comment:
Why not a `[Platform: String]` instead of a list of nullable strings?

Reply:
Also, maybe we could use a stricter type instead of String?

Reply:
Any given target can have a minimum deployment target set for each platform that exists, but only one platform can be set.  using an array would allow something like `[(.iOS, "13.0"), (.iOS, "14.0")]` which this approach prevents at the compiler level.  I used string because i was following the original `DeploymentTarget` pattern (open to changing this but it minimizes overall codechanges in the codebase).  And they're all optional because they aren't required by Xcode (missing versions get a default provided by that Xcode version) and they can all also be set via xcconfig files or build settings.